crates/mir/src/lower/types.rs

```diff
@@ -8,9 +8,13 @@ use crate::{
 
 use fe_analyzer::namespace::{items as analyzer_items, types as analyzer_types};
 
-pub fn lower_type(db: &dyn MirDb, analyzer_ty: &analyzer_types::TypeId) -> TypeId {
+pub fn lower_type(db: &dyn MirDb, analyzer_ty: analyzer_types::TypeId) -> TypeId {
     let ty_kind = match analyzer_ty.typ(db.upcast()) {
-        analyzer_types::Type::Base(base) => lower_base(&base),
+        analyzer_types::Type::SPtr(inner) => {
+            // TODO: lower to SPtr(lower_type(db, inner).
```

Comment:
I attempted this, but ended up with an SPtr(SPtr(Map)) type that I couldn't easily find the source of.

Reply:
what code ended up with the nested `SPtr`?
I tried the code below and it seems it returns collect MIR type `SPtr<Map<i32, i32>>`
```fe
contract Foo {
    m: Map<i32, i32>
    
    fn test(self) {
        self.m
    }
}
```
![graphviz (2)](https://user-images.githubusercontent.com/6376004/182858383-3a58569e-8f54-4fea-ba65-b2507d4e38bd.svg)